File: practice-projects/gin-ddd/internal/interfaces/vo/order/order_response.go
package order

import (
	orderDTO "gin-ddd/internal/application/dto/order"
	"time"
)

// OrderResponse 订单响应
type OrderResponse struct {
	ID          int64     `json:"id"`
	OrderNo     string    `json:"order_no"`
	UserID      int64     `json:"user_id"`
	TotalAmount float64   `json:"total_amount"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// FromOrderDTO 将订单DTO转换为响应VO
func FromOrderDTO(dto *orderDTO.OrderDTO) *OrderResponse {
	if dto == nil {
		return nil
	}
	return &OrderResponse{
		ID:          dto.OrderID,
		OrderNo:     dto.OrderNo,
		UserID:      dto.UserID,
		TotalAmount: dto.TotalAmount,
		Status:      dto.Status,
		CreatedAt:   dto.CreatedAt,
		UpdatedAt:   dto.UpdatedAt,
	}
}

// FromOrderDTOs 批量将订单DTO转换为响应VO
func FromOrderDTOs(dtos []*orderDTO.OrderDTO) []*OrderResponse {
	responses := make([]*OrderResponse, 0, len(dtos))
	for _, dto := range dtos {
		responses = append(responses, FromOrderDTO(dto))
	}
	return responses
}

File: practice-projects/nestjs-ddd/src/interfaces/http/order/vo/order.response.ts
import { ApiProperty } from '@nestjs/swagger';
import { OrderDto } from '../../../../application/order/dto/order.dto';
import { OrderStatus } from '../../../../domain/order/model/order.entity';

export class OrderResponse {
  @ApiProperty()
  id!: number;

  @ApiProperty({ name: 'order_no' })
  order_no!: string;

  @ApiProperty({ name: 'user_id' })
  user_id!: number;

  @ApiProperty({ name: 'total_amount' })
  total_amount!: number;

  @ApiProperty({ enum: OrderStatus })
  status!: OrderStatus;

  @ApiProperty({ name: 'created_at' })
  created_at!: Date;

  @ApiProperty({ name: 'updated_at' })
  updated_at!: Date;

  static fromDto(dto: OrderDto): OrderResponse {
    const vo = new OrderResponse();
    vo.id = dto.id;
    vo.order_no = dto.orderNo;
    vo.user_id = dto.userId;
    vo.total_amount = dto.totalAmount;
    vo.status = dto.status;
    vo.created_at = dto.createdAt;
    vo.updated_at = dto.updatedAt;
    return vo;
  }

  static fromDtos(dtos: OrderDto[]): OrderResponse[] {
    return dtos.map((d) => OrderResponse.fromDto(d));
  }
}
